sdcm/utils.py

```diff
@@ -176,16 +176,32 @@ def remove_comments(data):
 
 
 class S3Storage(object):
+
+    _bucket = None
+
+    @classmethod
+    def _get_bucket(cls, bucket_name):
+        if cls._bucket and cls._bucket.name == bucket_name:
+            return cls._bucket
+        return boto3.resource('s3').Bucket(name=bucket_name)
+
+    @classmethod
+    def find_files_by_testid_in_bucket(cls, bucket_name, testid):
+        files = []
+        bucket = cls._get_bucket(bucket_name)
+        for obj in bucket.objects.filter(Prefix=testid):
+            files.append(obj.key)
+        return files
+
     @staticmethod
-    def generate_url(file_path):
-        job_name = get_job_name()
+    def generate_url(file_path, prefix=''):
```

Comment:
name prefix is misleading, rename to smth. like
`prefix` -> `dir_to_store`

Reply:
S3 api use the prefix to filter objects in buckets

Reply:
Rewritten